doc/dev/index.rst

```diff
@@ -17,6 +17,7 @@ Here we document internal components of Arbor. These pages can be useful if you'
    cable_cell
    cell_groups
    communication
+   external
```

Comment:
where is this file?

Reply:
Currently, this is all covered in the _user_ docs, since I expect _external_ developers to come to look at this. Thus, I ended up not adding this.